src/window.rs

```diff
@@ -439,8 +439,15 @@ pub fn ui_loop() {
                     ..
                 } => {
                     keycode = received_keycode;
+                    keytexts.push((keycode, None));
+                }
+                Event::TextInput { text, .. } => {
```

Comment:
Do we know if textinput always happens after the keycode event?

Reply:
It appears that way. Though I place this line here in order to catch things like backspace and enter. It causes a double up with characters.

Reply:
my only concern is that sometimes sdl works differently on other platforms. You're probably right, but this may come back to bite us later